src/SUMMARY.md

```diff
@@ -60,5 +60,6 @@
 - [Contributing](contributing/README.md)
    - [Pull request checklist](contributing/pr_checklist.md)
    - [How to add operations to Parsec](contributing/adding_new_operation_how_to.md)
+   - [Upstreaming guide](contributing/upstreaming.md)
```

Comment:
Is "upstreaming" the correct word here? I would think more of "publishing" or "releasing"? "upstreaming" for me refers to pushing anything to the public, ie making pull-requests.

Reply:
I can change that - to me "upstreaming" is more general, moving something to somewhere higher in the consumption/distribution chain: local copy -> own fork -> original repo -> package manager. When other people use our code, they pull it from crates.io, not from GitHub (generally)

Reply:
(actually, it seems "publishing" is derived from the Latin word for "making public", I guess they're all kinda saying the same thing in different words)

Reply:
It's ok then!

Reply:
I think I agree with Hugues here - I think "upstreaming" is the wrong term, and this could be quite confusing to people who are just making a regular PR into master without any complex interface or dependency updates. I would call it a versioning guide, or a component versioning guide, or a versioning and dependency management guide.

Reply:
Also think that we should refer to the pieces as "components" rather than "projects".